src/pcapi/routes/native/v1/favorites.py

```diff
@@ -131,19 +132,20 @@ def create_favorite(user: User, body: serializers.FavoriteRequest) -> serializer
     if settings.MAX_FAVORITES:
         if Favorite.query.filter_by(user=user).count() >= settings.MAX_FAVORITES:
             raise ApiErrors({"code": "MAX_FAVORITES_REACHED"})
-    with transaction():
-        offer = Offer.query.filter_by(id=body.offerId).first_or_404()
-        favorite = Favorite.query.filter(Favorite.offerId == body.offerId, Favorite.userId == user.id).one_or_none()
 
-        if not favorite:
+    offer = Offer.query.filter_by(id=body.offerId).first_or_404()
+    try:
+        with transaction():
             favorite = Favorite(
                 mediation=offer.activeMediation,
                 offer=offer,
                 user=user,
             )
             db.session.add(favorite)
             db.session.flush()
-        return serializers.FavoriteResponse.from_orm(favorite)
+    except exc.IntegrityError:
+        favorite = Favorite.query.filter(Favorite.offerId == body.offerId, Favorite.userId == user.id).one_or_none()
```

Comment:
(détail / style)

```suggestion
        favorite = Favorite.query.filter_by(offerId=body.offerId, userId=user.id).one_or_none()
```

"plus court c'est mieux"